packages/x-components/src/components/snippet-callbacks.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <NoElement>
+    <GlobalXBus v-on="eventListeners" />
+  </NoElement>
+</template>
+
+<script lang="ts">
+  import { Component, Inject } from 'vue-property-decorator';
+  import Vue from 'vue';
+  import { map } from '../utils/object';
+  import { WireMetadata } from '../wiring';
+  import { SnippetConfig, XEventListeners } from '../x-installer/api/api.types';
+  import GlobalXBus from './global-x-bus.vue';
+  import { NoElement } from './no-element';
+
+  /**
+   * This component subscribes to any {@link XEvent} with a custom callbacks provided by the snippet
+   * configuration.
+   *
+   * @public
+   */
+  @Component({
+    components: { NoElement, GlobalXBus }
+  })
+  export default class SnippetCallbacks extends Vue {
+    /**
+     * Injects {@link SnippetConfig} provided by an ancestor as snippetConfig.
+     *
+     * @internal
+     */
+    @Inject('snippetConfig')
+    public snippetConfig!: SnippetConfig;
+
+    /**
+     * It maps all the callbacks provided by the snippetConfig and adds an emit to each one.
+     *
+     * @returns The event listeners with the {@link XEventsTypes.SnippetCallbackExecuted} emit in
+     * the callback.
+     *
+     * @internal
+     *
+     */
+    protected get eventListeners(): XEventListeners {
+      const { callbacks } = this.snippetConfig;
+      return callbacks
```

Comment:
If I'm understanding the `map` function correctly I think that you don't need to check for `callbacks` to be empty or not. If it's empty `map` will return an empty object and the function will return it in the correct type (might we invent, will test)

Reply:
The `callbacks` property could be definitely empty but it could also come as `undefined`. That is why I added the condition. At first I added a, empty object as a default value in the deconstruction but I think it is better to avoid the execution of `map` instead of forcing it with a default.